godot-core/src/builtin/plane.rs

```diff
@@ -105,16 +107,134 @@ impl Plane {
+        Some(result / denom)
+    }
+
+    /// Finds the intersection point of the plane with a ray.
+    ///
+    /// Returns the intersection point of a ray consisting of the position `from` and the direction 
+    /// normal `dir` with the `self` `Plane` enclosed in `Some`. If no intersection is found, `None`
+    /// will be returned.
+    #[inline]
+    pub fn intersects_ray(&self, from: Vector3, dir: Vector3) -> Option<Vector3> {
+	    let denom: real = self.normal.dot(dir);
+        if is_zero_approx(denom) {
+            return None;
+        }
+        let dist: real = (self.normal.dot(from) - self.d) / denom;
+        if dist > CMP_EPSILON {
+            return None;
+        }
+        Some(from - dir * dist)
+    }
+
+    /// Finds the intersection point of the plane with a line segment.
+    /// 
+    /// Returns the intersection point of a segment from position `from` to position `to` with the 
+    /// `self` `Plane` enclosed in `Some`. If no intersection is found, `None` will be returned.
+    #[inline]
+    pub fn intersects_segment(&self, from: Vector3, to: Vector3) -> Option<Vector3> {
+        let segment: Vector3 = from - to;
+        let denom: real = self.normal.dot(segment);
+        if is_zero_approx(denom) {
+            return None;
+        }
+        let dist: real = (self.normal.dot(from) - self.d) / denom;
+        if dist < -CMP_EPSILON || dist > (1.0 + CMP_EPSILON) {
+            return None;
+        }
+        Some(from - segment * dist)
+    }
+
+    /// Finds whether the two planes are approximately equal.
+    ///
     /// Returns `true` if the two `Plane`s are approximately equal, by calling `is_equal_approx` on
     /// `normal` and `d` or on `-normal` and `-d`.
-    ///
-    /// _Godot equivalent: `Plane.is_equal_approx()`_
     #[inline]
     pub fn is_equal_approx(&self, other: &Self) -> bool {
         (self.normal.is_equal_approx(other.normal) && is_equal_approx(self.d, other.d))
             || (self.normal.is_equal_approx(-other.normal) && is_equal_approx(self.d, -other.d))
     }
 
+    /// Finds whether the plane is finite or not.
+    ///
+    /// Returns `true` if the `self` `Plane` is finite by calling `is_finite` on `normal` and `d`. 
+    #[inline]
+    pub fn is_finite(&self) -> bool {
+        self.normal.is_finite() && self.d.is_finite()
+    }
+
+    /// Finds if a point is located above or below the plane.
+    ///
+    /// Returns `true` if `point` is located above the `self` `Plane`.
+    #[inline]
+    pub fn is_point_over(&self, point: Vector3) -> bool {
+        self.normal.dot(point) > self.d
+    }
+
+    /// Creates a normalized copy of the plane.
+    ///
+    /// Returns a normalized copy of the `self` `Plane`.
+    #[inline]
+    pub fn normalized(&self) -> Self {
+        Plane::new(self.normal.normalized(), self.d)
+    }
+
+    /// Projects a point orthogonally to the plane.
+    ///
+    /// Returns a orthogonal of the variable `point` into a point in the `self` `Plane`.
```

Comment:
This is not very clearly worded. Returns (an) orthogonal what? a point? points cant really be orthogonal to other points.